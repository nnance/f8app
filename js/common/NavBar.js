

import React from 'react';
import {
  Text,
  View,
  StyleSheet,
  TouchableOpacity,
  Image,
  Dimensions,
} from 'react-native';

import { styles as commonStyles, NAV_BAR_HEIGHT, STATUS_BAR_HEIGHT } from './styles';

export const HEIGHT = NAV_BAR_HEIGHT + STATUS_BAR_HEIGHT;

const styles = StyleSheet.create({
  container: {
    paddingTop: STATUS_BAR_HEIGHT,
    height: NAV_BAR_HEIGHT,
    width: Dimensions.get('window').width,
    backgroundColor: 'white',
    justifyContent: 'space-between',
    flexDirection: 'row',
    alignItems: 'center',
  },
  titleMenu: {
    flex: 4,
    alignItems: 'center',
  },
  leftMenu: {
    flex: 1,
    paddingLeft: 10,
    alignItems: 'flex-start',
  },
  rightMenu: {
    flex: 1,
    paddingRight: 10,
    alignItems: 'flex-end',
  },
  navText: {
    fontSize: 22,
    fontWeight: 'bold',
  },
});

export default class NavBar extends React.Component {
  renderTitle() {
    return (
      <Text style={[styles.navText, this.props.titleTextStyle]}>{this.props.title}</Text>
    );
  }

  render() {
    return (
      <View style={[styles.container, this.props.containerStyle]}>
        <View style={[styles.leftMenu, this.props.leftMenuStyle]}>
          {
            this.props.renderLeftMenu ? this.props.renderLeftMenu() : null
          }
        </View>
        <View style={[styles.titleMenu, this.props.titleStyle]}>
          {
            this.props.renderTitle ? this.props.renderTitle() : this.renderTitle()
          }
        </View>
        <View style={[styles.rightMenu, this.props.rightMenuStyle]}>
          {
            this.props.renderRightMenu ? this.props.renderRightMenu() : null
          }
        </View>
      </View>
    );
  }
}

export class NavBarWithButton extends React.Component {
  constructor(...args) {
    super(...args);
    this.renderBackButton = this.renderBackButton.bind(this);
  }

  renderBackButton() {
    return (
      <TouchableOpacity onPress={this.props.onBackPress}>

              <View style={{ width: 40, height: 40, flex: 0, marginTop: 0, justifyContent: 'center', alignItems: 'center' }}>
          <Image style={commonStyles.navBarIcon} source={require('../assets/common/white-back-button.png')} />
        </View>
        </TouchableOpacity>
    );
  }

  render() {
    return (<NavBar
      renderLeftMenu={this.renderBackButton}
      {...this.props}
    />);
  }
}

export class NavBarWithPinkButton extends React.Component {
  constructor(...args) {
    super(...args);
    this.renderBackButton = this.renderBackButton.bind(this);
  }

  renderBackButton() {
    return (
      <TouchableOpacity onPress={this.props.onBackPress}>
        <View style={{ height: 30, width: 40, flex: 0, marginTop: 0, justifyContent: 'center', alignItems: 'center' }}>
          <Image style={commonStyles.navBarIcon} source={require('../assets/common/pink-back-button.png')} />
        </View>
        </TouchableOpacity>
    );
  }

  render() {
    return (<NavBar
      renderLeftMenu={this.renderBackButton}
      {...this.props}
    />);
  }
}
