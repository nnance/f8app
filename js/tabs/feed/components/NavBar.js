import React from 'react';

import { NavBarWithButton } from '../../../common/NavBar';

export { HEIGHT } from '../../../common/NavBar';

export default class NavBar extends React.Component {
  render() {
    return (
      <NavBarWithButton
        {...this.props}
      />
    );
  }
}
