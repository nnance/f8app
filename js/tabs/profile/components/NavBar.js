

import React from 'react';
import {
  StyleSheet,
  TouchableOpacity,
} from 'react-native';

import { NavBarWithPinkButton } from '../../../common/NavBar';

import { colors } from '../../../common/styles';

const styles = StyleSheet.create({
  container: {
    borderBottomWidth: 1,
    borderColor: colors.greyBorder,
  },
  backButton: {
    height: 20,
    resizeMode: 'contain',
  },
  titleText: {
    fontSize: 22,
    fontWeight: 'bold',
    color: colors.textPink,
  },
});

export default class NavBar extends React.Component {
  constructor(...args) {
    super(...args);
    this.renderRightMenu = this.renderRightMenu.bind(this);
  }

  renderRightMenu() {
    return (
      <TouchableOpacity style={styles.rightMenu} onPress={this.props.onRightPress}>
        {
          this.props.renderRightMenu ? this.props.renderRightMenu() : null
        }
      </TouchableOpacity>
    );
  }

  render() {
    return (
      <NavBarWithPinkButton
        {...this.props}
        containerStyle={styles.container}
        renderRightMenu={this.renderRightMenu}
        title={this.props.title}
        titleTextStyle={styles.titleText}
      />
    );
  }
}
