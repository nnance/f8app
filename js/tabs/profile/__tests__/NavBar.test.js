import React from 'react';
import {
  Text,
  TouchableOpacity,
  View,
} from 'react-native';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import NavBar from '../components/NavBar';

describe('<NavBar/>', () => {
  it('render backButton by default', () => {
    const tree = renderer.create(<NavBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('can custom rightMenu', () => {
    const dump = shallow(<NavBar renderRightMenu={() => <Text>RButton</Text>} />);
    const wrapper = shallow(<View>{dump.instance().renderRightMenu()}</View>);
    expect(wrapper.find('Text').length).toBe(1);
  });

  it('handle onPress', () => {
    const leftFn = jest.fn();
    const rightFn = jest.fn();
    const dump = shallow(
      <NavBar
        renderRightMenu={() => <Text>RButton</Text>}
        onLeftPress={leftFn}
        onRightPress={rightFn}
      />,
    );
    const wrapper = shallow(<View>{dump.instance().renderRightMenu()}</View>);
    const touchables = wrapper.find(TouchableOpacity);
    expect(touchables.length).toBeGreaterThan(0);
    touchables.at(0).simulate('press');
    expect(leftFn).not.toBeCalled();
    expect(rightFn).toBeCalled();
  });
});
