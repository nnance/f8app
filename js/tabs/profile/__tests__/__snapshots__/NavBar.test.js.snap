exports[`<NavBar/> render backButton by default 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexDirection": "row",
        "height": 60,
        "justifyContent": "space-between",
        "paddingTop": 15,
        "width": 750,
      },
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#eeeeee",
      },
    ]
  }>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "paddingLeft": 10,
        },
        undefined,
      ]
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Image
        source={1}
        style={
          Object {
            "height": 20,
            "resizeMode": "contain",
          }
        } />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 4,
        },
        undefined,
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 22,
            "fontWeight": "bold",
          },
          Object {
            "color": "rgb(249, 65, 92)",
            "fontSize": 22,
            "fontWeight": "bold",
          },
        ]
      } />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "paddingRight": 10,
        },
        undefined,
      ]
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined} />
  </View>
</View>
`;
